### ISSUE TITLE
Add option for compact layout

### ISSUE THREAD

**Issue by soraphis** [files-community/Files#1940]
**Is your feature request related to a problem? Please describe.**
Its annoying that so much space is wasted in comparison to the default explorer

**Describe the solution you'd like**
An option to reduce the font size

**Describe alternatives you've considered**
Changing the windows scale seems not to work on a per-app basis on UWP apps

**Additional context**
comparison screenshot:
left the old explorer showing 27 items, right the files-uwp showing 16 on the same window size. 
![image](https://user-images.githubusercontent.com/3066582/93582486-d7471700-f9a2-11ea-816e-4b9a13ce8001.png)

| Capability | Priority |
| :---------- | :------- |
| This proposal will add a setting to turn on the compact layout | Must |
| This option will use the compact styles for the sidebar and details layout | Must |

**Comment by duke7553** [comment]
Some of the work that is currently in the proposal drafting stage should pave the way for density adjustments to the List control. We adjusted the default font size several months ago, but recognize the padding isn't as suitable as it could be for some users.

**Comment by Jaiganeshkumaran** [comment]
Not needed. Already the size is so small and reducing it will make it unusable with touch. The File Explorer is an old application designed when you only used your PC on a desk with a mouse so that's why it's too small. This application should also work with touch and having it too small will be bad.

**Comment by generalguy41** [comment]
@Jaiganeshkumaran I disagree that it's not needed. As a primarily desktop user, I think that there should be an option to have a "compact" mode. By default, it could use the more spaced out mode, but I'd like to see the option available at least.

**Comment by Jaiganeshkumaran** [comment]
@generalguy41 I'm a desktop user too and don't have a touch screen but it shouldn't be that small

**Comment by generalguy41** [comment]
@Jaiganeshkumaran Agree to disagree, I guess. I would much rather have an option to show more items in the window and have a more compact view. Ultimately the decision isn't up to me, but I'd seriously hope it would be taken it into consideration for more "power users" I guess.

**Comment by perplexityjeff** [comment]
Searched as I am having this issue as well. I find that from a productivity standpoint that the old explorer shows you so much more information at a glance. For me at least an option to tweak this to my liking would be fantastic. Either font settings and/or the padding.

**Comment by trama** [comment]
I also agree with the fact there should be an option to have a more dense view for productivity when in desktop mode. I mean, I totally understand the reasoning of @Jaiganeshkumaran about touch, but having a set of users that interact through touch screens -hence needing larger controls- should not impact on the productivity needs of desktop/power users. Hence, the request for an /option/, not a single choice. This should not be regarded as something so much strange, imho. For instance, in OSX the Finder applications has really narrow spaces (borders and paddings) and no one complain about that, even more it is one of the most loved productivity app on a Mac... 
to be honest, I think that Files has all the potentialities (given the current development point) to be even better (For instance, Files has a native dual pane, multiple tabs is more efficient), but being more compact in desktop environments would be fantastic.

I would like to point also to the fact winUI controls already foreseen a "compact" mode in their default resources. Please, refer to this page:
https://docs.microsoft.com/it-it/windows/uwp/design/style/spacing

The point is to add (controlled by an option) this line 
<ResourceDictionary Source="ms-appx:///Microsoft.UI.Xaml/DensityStyles/Compact.xaml" />
and this should already provide a more compact view of the different controls, given that the whole Files app, correctly, refers to the default ThemeResource values.

There is an example already provided by Microsoft about this: https://github.com/Microsoft/Xaml-Controls-Gallery

I hope there should be an option soon. You have carried out a fantastic work so far, and Files is totally stable now...so that I would use it for each task, provided it is as usable as other for power users.

**Comment by Gabryxx7** [comment]
I feel the same about having the option to customize it at least a few pre-defined choices.
I wrote my own workaround so I can easily adjust padding, margins and font size.
Compiling takes a few minutes so I did not want to go through that everytime I had to make a small change to the padding for the sidebar or the list view so I implemented a quick theme-based workaround. I pretty much just made it look for the padding values in the theme xaml file instead of using pre-defined ones.


In my `sample1.xaml` file I added these few lines in the default theme:

```xaml
                            <Thickness x:Key="ExpanderContentPadding">20,0,0,0</Thickness>
                            <Thickness x:Key="ExpanderMinHeight">1</Thickness>
                            <Thickness x:Key="GridViewHeaderPadding">12,8,12,0</Thickness>
                            <Thickness x:Key="GridViewHeaderMinHeight">0</Thickness>
                            <Thickness x:Key="GridViewHeaderItemMinHeight">0</Thickness>
                            <Thickness x:Key="ListViewRowPadding">8,0,8,8</Thickness>
                            <Thickness x:Key="ListViewHeaderGridHeight">38</Thickness>
                            <Thickness x:Key="ListedItemPadding">10,0</Thickness>
                            <Thickness x:Key="ListedItemHeight">32</Thickness>
                            <Thickness x:Key="ListViewMinHeight">1</Thickness>
                            <Thickness x:Key="ListViewHeight">36</Thickness>
                            <Thickness x:Key="ListViewItemMinHeight">1</Thickness>
                            <Thickness x:Key="ListViewHeaderItemPadding">12,8,12,0</Thickness>
```

I also added these lines to `default`, `dark` and `light` in the `App.xaml` file.

After doing that I changed a few lines to make sure the program was referencing those value from the theme resources.
In `SettingsBlockControl.xaml`
```xaml
            <Setter Property="MinHeight" Value="{ThemeResource ExpanderMinHeight}" />
            <Setter Property="Padding" Value="{ThemeResource ExpanderContentPadding}"  />
            ...
            <Setter Property="Padding" Value="{ThemeResource ExpanderContentPadding}"  />
            <Setter Property="MinHeight" Value="{ThemeResource ExpanderMinHeight}" />
```

In `DetailsLayoutBrowser.xaml`
```xaml
                <Setter Property="Padding" Value="{ThemeResource GridViewHeaderPadding}" />
            ...
                    <ListView.Header>
                        <Grid
                            x:Name="HeaderGrid"
                            Height="38"
                            Height="{ThemeResource ListViewHeaderGridHeight}"
                            Margin="16,0,0,0"
                            Padding="0"
                            Background="{ThemeResource FileBrowserHeaderBackgroundBrush}"
            ...
                             <ContentPresenter Padding="{ThemeResource ListViewRowPadding}" Content="{TemplateBinding Header}" />
            ...
                    <ListView.ItemContainerStyle>
                        <Style BasedOn="{StaticResource DefaultListViewItemStyle}" TargetType="ListViewItem">
                            <Setter Property="HorizontalContentAlignment" Value="Stretch" />
                            <Setter Property="MinHeight" Value="{ThemeResource ListViewMinHeight}" />
                            <Setter Property="Height" Value="{ThemeResource ListViewHeight}" />
                        </Style>
                    </ListView.ItemContainerStyle>
```

You can now change these values in your `sample1.xaml` file, this is what mine looks like
```xaml
            <Thickness x:Key="ExpanderContentPadding">20,0,0,0</Thickness>
            <Thickness x:Key="ExpanderMinHeight">1</Thickness>
            <Thickness x:Key="GridViewHeaderPadding">12,8,12,0</Thickness>
            <Thickness x:Key="GridViewHeaderMinHeight">0</Thickness>
            <Thickness x:Key="GridViewHeaderItemMinHeight">0</Thickness>
            <Thickness x:Key="ListViewRowPadding">4,0,4,4</Thickness>
            <Thickness x:Key="ListViewHeaderGridHeight">38</Thickness>
            <Thickness x:Key="ListViewRowPadding">5,0,5,5</Thickness>
            <Thickness x:Key="ListedItemPadding">10,0</Thickness>
            <Thickness x:Key="ListedItemHeight">32</Thickness>
            <Thickness x:Key="ListViewMinHeight">1</Thickness>
            <Thickness x:Key="ListViewHeight">25</Thickness>
            <Thickness x:Key="ListViewItemMinHeight">1</Thickness>
            <Thickness x:Key="ListViewHeaderItemPadding">12,8,12,0</Thickness>
            <Thickness x:Key="SplitViewLeftBorderThemeThickness">20</Thickness>
            <Thickness x:Key="SplitViewPaneRootCornerRadius">50</Thickness>
            <Thickness x:Key="NavigationViewContentGridBorderThickness">20,0,0,0</Thickness>
```
And this is the end result!
![test](https://user-images.githubusercontent.com/6282055/127130705-d3e81585-a27c-4622-b861-365c8c163f76.PNG)
![test2](https://user-images.githubusercontent.com/6282055/127131005-ddb20138-ca44-436e-981b-c3fbef488b21.PNG)

You can follow the same process for the font size. I believe the key can be a `Thickness` as well as that's just a `double` value container. Then reference it in the code wherever the fontsize is used

**Comment by samsong** [comment]
I support this too. If it's not too much hassle I would suggest a zoom +- button like chrome browser has.
This could increment up down the font size & spacing by preset values approximately 3 steps lower or 3 steps higher.

**Comment by idkman2021** [comment]
sooo will this be fixed?

**Comment by Gabryxx7** [comment]
Hey, I am rather busy lately but I could try and implement the fix I mentioned above in the next couple of weeks. Code might not be amazing but it would be a start :)

**Comment by yaichenbaum** [comment]
> Hey, I am rather busy lately but I could try and implement the fix I mentioned above in the next couple of weeks. Code might not be amazing but it would be a start :)

That would be great! I guess we should figure out which surfaces should use the compact styles. My initial thoughts are the sidebar, all the different layout modes and context menus.

**Comment by Andy-2732** [comment]
In Win11 File Explorer this is a thing, would be great to have this in Files!
![image](https://user-images.githubusercontent.com/92577745/149219707-8b367be3-f4d6-4b30-8b3a-eb49ebfb3599.png)

**Comment by yaichenbaum** [comment]
> Hey, I am rather busy lately but I could try and implement the fix I mentioned above in the next couple of weeks. Code might not be amazing but it would be a start :)

@Gabryxx7 just wanted to follow up and see if you had a chance to look into this.

**Comment by Gabryxx7** [comment]
Yes I did started working on it!
I barely use WPF and C# so I hit a bit of a road block.
I would like to bind one of the sliders in the screenshot to the `Margin` property of an `ItemContainerStyle` inside `DetailsLayoutBrowser.xaml`.

```xaml
  <ListView.ItemContainerStyle>
                <Style BasedOn="{StaticResource DefaultListViewItemStyle}" TargetType="ListViewItem">
                    <Setter Property="HorizontalContentAlignment" Value="Stretch" />
                    <Setter Property="MinHeight" Value="{ThemeResource ListViewMinHeight}" />
                    <Setter Property="Height" Value="{ThemeResource ListViewHeight}" />
                    <Setter Property="Margin" Value="{Binding SpacingSize, Mode=OneWay}" />
                </Style>
   </ListView.ItemContainerStyle>
```

The issue I'm facing is that even when clicking the refresh button, the margin property does not seem to be updated, most likely because the style has already been applied and therefore the bound property `SpacingSize` won't be read anymore.

Anyone knows how to force a refresh whenever the slider's value changes?

**Comment by Gabryxx7** [comment]
> @samsong what happens if you set the height of the list view item to auto and bind the height setting to the grid within the list item?

I could try that. It'd be nicer if the height was set to Auto and calculated through margin and padding but as of now only properties bound to the ListedItem (as opposed to the style/setter) seem to get refreshed.

**Comment by Gabryxx7** [comment]
Hi everyone!
I submitted PR #7941 to add settings for customizing Padding and Spacing in all Layouts (not just Details).

It took me a while to figure out how all the parts are connected but I came up with a solution at the most general level, implemented by changind `Padding `and `Margin `of `ListViewItem `(or `GridViewItem`) from both `ModernShellPage `and `ColumnShellPage`.

The reason being that they contain any Layout that implements `BaseLayout`, stored in `ContentPage`.
Moving the sliders updates Spacing and Padding live in any layout!

**Comment by takanuva15** [comment]
@Gabryxx7 you are a legend sir!

Quick question: Do the changes also reduce spacing within the right-click context menu? (Windows 11 decided to overdo the spacing in the context menus too)

**Comment by Gabryxx7** [comment]
> @Gabryxx7 you are a legend sir!
> 
> Quick question: Do the changes also reduce spacing within the right-click context menu? (Windows 11 decided to overdo the spacing in the context menus too)

No they don't, they only affect the Browser page. I'll have a look at how to implement the changes on a more general level to also include the sidebar and the context menu!

**Comment by yaichenbaum** [comment]
@Gabryxx7 considering the fact that this will eventually affect the context menu and sidebar, do you think the options should be moved to the settings dialog?

**Comment by Gabryxx7** [comment]
I thought about it actually, but I thought it'd be much nicer to see the change live from the little flyout menu on the navbar. And the way I see it, these two sliders affect the file browser just like the layout radio buttons and the "Show hidden files/ Show extension" toggles.

Shall we maybe have two different settings now that I'll try to add those for menu and sidebar? So one set of sliders for the file browser and one for the UI in general (so sidebar and menu at the moment)?

On a side note, @yair100 I was thinking of adding another slider for icons/font scaling, is it something we are interested in? Or is that too much customization? 

From my point of view, these should NOT be part of a theme. Sizing is something that changes according to screen size and one's preferred work environment setting.

**Comment by yair100** [comment]
> I thought about it actually, but I thought it'd be much nicer to see the change live from the little flyout menu on the navbar. And the way I see it, these two sliders affect the file browser just like the layout radio buttons and the "Show hidden files/ Show extension" toggles.
Shall we maybe have two different settings now that I'll try to add those for menu and sidebar? So one set of sliders for the file browser and one for the UI in general (so sidebar and menu at the moment)?

I'm afraid that will make the options too complex, and users would be confused when they don't match up.

> On a side note, @yair100 I was thinking of adding another slider for icons/font scaling, is it something we are interested in? Or is that too much customization?

Do you think we can automatically change the scale together with the padding and spacing?

**Comment by takanuva15** [comment]
> I thought about it actually, but I thought it'd be much nicer to see the change live from the little flyout menu on the navbar. And the way I see it, these two sliders affect the file browser just like the layout radio buttons and the "Show hidden files/ Show extension" toggles.

@Gabryxx7 @yair100 Just to give my 2 cents - I think what makes the most sense from a clean & simple UX perspective is to just have a single option in the Navbar Setting dialog. Call it something like "Density" and put a dropdown (like an HTML select element) there with 3 options "Compact", "Default", and "Comfortable". (Obviously "compact" would have minimum space/padding, "default" would be set to what it is currently, and "Comfortable" would increase space/padding over the default but not all the way to the max). This is actually similar to what Gmail already has for their inbox-compactness adjustment, which I've found works very well for the majority of users:

![image](https://user-images.githubusercontent.com/6986426/151099960-4545bdcf-c9af-40a2-b450-b4feca441ea0.png)

The sliders to fine-tune the padding/spacing should migrate to the Settings dialog that @yair100 was talking about, so that it's there for the few users that may not like any of the 3 defaults. 

The font and icons would have a preset scale with the 3 default options mentioned above (so that if a user selects "compact", then the font/icons would shrink slightly, and if they selected "comfortable", the font/icons would get slightly larger.) After that, any further options to set a custom icon or font size would move below the sliders within the Settings dialog (that @yair100 was talking about), so again users that may not like the defaults can simply type their desired pixel/scale size.

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
I believe that configuration should be possible within 1 separate window, because much of the present configuration is immediately visually obvious, and duplication and inconsistent placement of configurative functionality shall not merely unnecessarily clutter the interface, but confuse new users and be bothersome for all else.

**Comment by yodaluca23** [comment]
This needs to be added ASAP if it's not I may be forced to uninstall it this "Issue" has been open for over 2 years and it's still not implemented!?!?!?!?

Something just like This: https://github.com/files-community/Files/issues/1940#issuecomment-1016612717 
would be perfect except also add it to the left column.

**Comment by yaichenbaum** [comment]
@yodaluca23 someone happens to be working on this but in general, we don't prioritize issues based on how long they've been open.

**Comment by nharrer** [comment]
This was also the first thing that irked me. I see 23 items in Files. And 43 items in File Explorer. That's almost twice as much. That's a lot of wasted white pixels on the screen. 
Also, if you implement something, don't forget the left side and the menus. Get rid of the "more..." in the menus and display everything in one menu. That was one of the most criticized mistakes in Windows 11. You should not do this.
Other then those little peanuts - an awesome tool. Great work.

**Comment by ChrisMooreGB** [comment]
Thanks @yair100 for merging my #8350 with this one - apologies for not locating this one myself. I don't know how difficult it is to implement the option of Wider/Tighter spacing - presumably not trivial as it's been on the wishlist for so long. For me, my current solution of two File Explorer windows on screen works better for me that Files with its wide line spacing. For the record I am overwhelmingly a Windows Desktop - rather than touch - user.

**Comment by yaichenbaum** [comment]
@ChrisMooreGB thank you for your understanding, there is an open pull request for this feature already so it will hopefully be available in one of the next updates.

**Comment by Gabryxx7** [comment]
Hey everyone! I started implementing these features in PR #7941 , unfortunately I got really busy with my PhD and some other mental health issues. Things are looking a bit better now so I'll deep dive into it again this weekend! I have to figure out how to link all the settings to the elements' properties and how to add the new settings to the menu but I had a rough idea of where to start :)

**Comment by yaichenbaum** [comment]
@Gabryxx7 I'm glad to hear that things are looking better for you, take all the time you need and if you have any questions, just let us know.

**Comment by Gabryxx7** [comment]
> @Gabryxx7 I'm glad to hear that things are looking better for you, take all the time you need and if you have any questions, just let us know.

Do you guys have a discord or any platform that I can contact you on about quick questions (e.g. where certain code is, which file to edit, code format etc...)? My username is also gabryxx7 on Discord (and everywhere else really).

**Comment by itsTyrion** [comment]
@Gabryxx7 
> My username is also gabryxx7 on Discord

FYI your username on Discord consists of the name AND the tag (like \#1234)

**Comment by swebervna** [comment]
> Not needed. Already the size is so small and reducing it will make it unusable with touch. The File Explorer is an old application designed when you only used your PC on a desk with a mouse so that's why it's too small. This application should also work with touch and having it too small will be bad.

I disagree, you could simply do what the current File Explorer and Windows UI already do on Windows 11, and that switches from compact to normal automatically when the laptop goes into tablet mode (for 2-in-1s). Besides, you could just manually switch it if you need to. More often people are using mouse versus touch, so features should prioritize the majority rather than the minority, but still give an option/solution for the minority.

**Comment by yaichenbaum** [comment]
This issue should be resolved with #8751

**Comment by yodaluca23** [comment]
> This issue should be resolved with #8751

So when will it be merged and the update be released??? Any ETA?

**Comment by soraphis** [comment]
Just checked out Version: 2.3.0.0 | OS Version: 10.0.22000.675

![chrome_7tGNJogUiW](https://user-images.githubusercontent.com/3066582/171926059-a4b9fb74-5006-4aa2-90e3-af85fa071b6c.png)
_left, win 11 explorer. on the right files compact mode_

without compact mode: 20 elements on screen. with compact mode 25 elements, win 11 explorer 34 elements

| explorer | elements visible | % |
| --------- | --------- | --------- |
| files no compact mode | 20 | 100% |
| files no compact mode | 25 | 125% |
| files no compact mode | 34 | 170% |
